### PR TITLE
Add What's new content for 7.12 release

### PR DESCRIPTION
**Pull request by EamonnTP**
### Summary

This PR adds the **What's new** content for the 7.12 release.

### HTML preview

https://observability-docs_461.docs-preview.app.elstc.co/guide/en/observability/7.12/whats-new.html

### Related issue

Closes https://github.com/elastic/observability-dev/issues/1397

### CODE REVIEW

File: docs/en/observability/whats-new.asciidoc [hunk 1]
```diff
@@ -1,21 +1,246 @@
...
+== Prometheus client for custom metrics in APM agents (Python only)
+
+[discrete]
+== Native support for OpenTelemetry (Experimental)
+
+OpenTelemetry is emerging as the standardized collection layer for observability data and is designed to enable
+organizations to instrument their apps in an open and vendor-neutral manner. Our support for OpenTelemetry is not new.
+In July 2020, we released the OpenTelemetry Collector exporter for Elastic that acts as a translation layer between OpenTelemetry agents and APM Server.
+
+[role="screenshot"]
```
Comment (bmorelli25) [inline docs/en/observability/whats-new.asciidoc:59]: `[role="screenshot"]` adds a box shadow to images. It makes sense for screenshots, but for diagrams like this, I think it's better to let them blend into the page. Example:

Screen Shot 2021-03-22 at 8 44 55 AM

---

Screen Shot 2021-03-22 at 8 45 00 AM


```suggestion
```

File: docs/en/observability/whats-new.asciidoc [hunk 2]
```diff
@@ -1,21 +1,246 @@
...
+image::images/otel-exporter-arch.png[OpenTelemetry Elastic exporter]
+
+In {minor-version}, we are pleased to announce the experimental native support of the OpenTelemetry protocol. This experimental integration is currently
+available for Elastic {observability} on self-managed deployments and available on {ecloud} and {ece} soon.
+
+This allows you to send collected telemetry data directly from your applications to APM Server. 
+Not only does this lower the effort to adopt OpenTelemetry into your existing Elastic architecture, but it also simplifies the architecture by removing
+the need to install and manage an extra component.
+
+[role="screenshot"]
```
Comment (bmorelli25) [inline docs/en/observability/whats-new.asciidoc:69]:
```suggestion
```

File: docs/en/observability/whats-new.asciidoc (hunk 1 above)
Comment (EamonnTP) [reply docs/en/observability/whats-new.asciidoc]: Good point. Thanks!

**Comment by tbragin** [comment]
@EamonnTP Happy to review. Could you update the preview when you get a chance? Sorry about late additions.
https://observability-docs_461.docs-preview.app.elstc.co/guide/en/observability/7.12/whats-new.html

**Comment by EamonnTP** [comment]
Thanks @tbragin The [latest preview](https://observability-docs_461.docs-preview.app.elstc.co/guide/en/observability/7.12/whats-new.html) is now available.

**Comment by tbragin** [comment]
@EamonnTP Looking good!

Is there a chance we can include the first level headers I had in the draft: Getting Started, Unified Analyst Experience, etc.? I'm hoping it will improve understanding for why we ordered individual features the way we did.

**Comment by EamonnTP** [comment]
Hi @tbragin Due to our doc build limitations, if we added the first-level headings we would lose the subheadings listed n the table of contents on the right hand side, as it doesn't support nested headings. Only the first-level headings would be listed.

**Comment by tbragin** [comment]
Ah, that's a bummer. Ok, let us leave it as is then! LGTM

**Review by tbragin — APPROVED** [review]
LGTM

**Pull request merged**